common/Unit.cpp

```diff
@@ -32,9 +32,9 @@
 #include <common/SigUtil.hpp>
 #include <common/StringVector.hpp>
 
-UnitKit *GlobalKit = nullptr;
-UnitWSD *GlobalWSD = nullptr;
-UnitTool *GlobalTool = nullptr;
+std::atomic<UnitKit *>GlobalKit = nullptr;
+std::atomic<UnitWSD *>GlobalWSD = nullptr;
+std::atomic<UnitTool *>GlobalTool = nullptr;
```

Comment:
We really need these as std::atomic ? - and GlobalTool too ? I'm surprised at the race this fixes - I would have preferred longer stack traces from the race in the changelog so we can see which one should really be serialized on the other and/or what order they should happen in.

Reply:
Yes of course we need it, as I ran into a data race which was already earmarked in 
`UnitWSD::DocBrokerDestroy` where GlobalWSD etc are set:

> // We could be called from either a SocketPoll (websrv_poll)
> // or from invokeTest (coolwsd main).

Then the static UnitWSD::get() or UnitKit::get() gets called
from e.g. coolwsd with potential old non-sync'ed data.

For this, I only had the assert SIGSEGV stacktrace which I have not saved sadly,
but to no surprise such assertion would occur every now and then w/o atomic.

(though I have stored my _sockePoll logs w/ backtraces to all data races from my earlier triage logs ...)

Reply:
SIGABRT (not SIGSEGV) due to assert as mentioned in the issue description. Sorry.

Reply:
The atomics here are unnecessary and most certainly don't fix anything. The reason is because we are already taking a (new) lock left and right.

We set these synchronously, at startup, when we have but one thread.

Unless something drastically changed that caused a regression, I find it incredible that we went for 9 years without seeing a single segfault until now.